iron-overlay-behavior.d.ts

```diff
@@ -0,0 +1,263 @@
+
+    /**
+     * Array of nodes that can receive focus (overlay included), ordered by `tabindex`.
+     * This is used to retrieve which is the first and last focusable nodes in order
+     * to wrap the focus for overlays `with-backdrop`.
+     *
+     * If you know what is your content (specifically the first and last focusable children),
+     * you can override this method to return only `[firstFocusable, lastFocusable];`
+     */
+    _focusableNodes(): any;
+    ready(): any;
+    attached(): any;
+    detached(): any;
+
+    /**
+     * Toggle the opened state of the overlay.
+     */
+    toggle(): any;
+
+    /**
+     * Open the overlay.
+     */
+    open(): any;
+
+    /**
+     * Close the overlay.
+     */
+    close(): any;
+
+    /**
+     * Cancels the overlay.
+     *
+     * @param event The original event
+     */
+    cancel(event?: Event|null): any;
+
+    /**
+     * Invalidates the cached tabbable nodes. To be called when any of the focusable
+     * content changes (e.g. a button is disabled).
+     */
+    invalidateTabbables(): any;
+    _ensureSetup(): any;
+
+    /**
+     * Called when `opened` changes.
+     */
+    _openedChanged(opened?: boolean): any;
+    _canceledChanged(): any;
+    _withBackdropChanged(): any;
+
+    /**
+     * tasks which must occur before opening; e.g. making the element visible.
+     */
+    _prepareRenderOpened(): any;
+
+    /**
+     * Tasks which cause the overlay to actually open; typically play an animation.
+     */
+    _renderOpened(): any;
+
+    /**
+     * Tasks which cause the overlay to actually close; typically play an animation.
+     */
+    _renderClosed(): any;
+
+    /**
+     * Tasks to be performed at the end of open action. Will fire `iron-overlay-opened`.
+     */
+    _finishRenderOpened(): any;
+
+    /**
+     * Tasks to be performed at the end of close action. Will fire `iron-overlay-closed`.
+     */
+    _finishRenderClosed(): any;
+    _preparePositioning(): any;
+    _finishPositioning(): any;
+
+    /**
+     * Applies focus according to the opened state.
+     */
+    _applyFocus(): any;
+
+    /**
+     * Cancels (closes) the overlay. Call when click happens outside the overlay.
+     */
+    _onCaptureClick(event: Event): any;
+
+    /**
+     * Keeps track of the focused child. If withBackdrop, traps focus within overlay.
+     */
+    _onCaptureFocus(event: Event): any;
+
+    /**
+     * Handles the ESC key event and cancels (closes) the overlay.
+     */
+    _onCaptureEsc(event: Event): any;
+
+    /**
+     * Handles TAB key events to track focus changes.
+     * Will wrap focus for overlays withBackdrop.
+     */
+    _onCaptureTab(event: Event): any;
+
+    /**
+     * Refits if the overlay is opened and not animating.
+     */
+    _onIronResize(): any;
+
+    /**
+     * Will call notifyResize if overlay is opened.
+     * Can be overridden in order to avoid multiple observers on the same node.
+     */
+    _onNodesChange(): any;
```

Comment:
is `_manager` property being omitted on purpose because it's private?
https://github.com/PolymerElements/iron-overlay-behavior/blob/master/iron-overlay-behavior.html#L132

Reply:
Yes, we don't emit private things.